src/main/java/ch/njol/skript/doc/Documentation.java

```diff
@@ -61,42 +60,33 @@ public class Documentation {
 	private static final Pattern CP_EMPTY_PARSE_MARKS_PATTERN = Pattern.compile("\\(\\)");
 	private static final Pattern CP_PARSE_TAGS_PATTERN = Pattern.compile("(?<=[(|\\[ ])[-a-zA-Z0-9!$#%^&*_+~=\"'<>?,.]*?:");
 	private static final Pattern CP_EXTRA_OPTIONAL_PATTERN = Pattern.compile("\\[\\(((\\w+? ?)+)\\)]");
-
-	private static boolean docsTemplateFound;
-	private static boolean generateUnsafeDocs;
-	private static final File docsTemplateDirectory = new File(Skript.getInstance().getDataFolder() + File.separator + "doc-templates");
-	private static final File docsDirectory = new File(Skript.getInstance().getDataFolder() + File.separator + "docs");
+	private static final File DOCS_TEMPLATE_DIRECTORY = new File(Skript.getInstance().getDataFolder() + File.separator + "doc-templates");
+	private static final File DOCS_OUTPUT_DIRECTORY = new File(Skript.getInstance().getDataFolder() + File.separator + "docs");
 
 	/**
 	 * Force register hooks even if their plugins are not present in the server
 	 */
-	public static final String FORCE_HOOKS_SYSTEM_PROPERTY = System.getProperty(("skript.forceregisterhooks"));
+	public static final String FORCE_HOOKS_SYSTEM_PROPERTY = System.getProperty("skript.forceregisterhooks");
 
-	public static boolean isDocsTemplateFound() {
-		return docsTemplateFound;
-	}
+	private static final boolean GENERATE_UNSAFE_DOCS = isDocsTemplateFound() && "true".equals(FORCE_HOOKS_SYSTEM_PROPERTY);
```

Comment:
I'm not sure if this should be a field, because it means for `GENERATE_UNSAFE_DOCS` it only works if the doc templates dir was found at startup, while for everywhere docs can be generated if the templates are found when requested

Reply:
Good point, will solve it 👍